### ISSUE TITLE
Dist-git MR is not closed when the source-git MR is closed

### ISSUE THREAD

**Issue by csomh** [packit/hardly#70]
Example:
- Source-git MR: https://gitlab.com/packit-service/src/systemd/-/merge_requests/2
- Dist-git MR: https://gitlab.com/packit-service/rpms/systemd/-/merge_requests/1

This should work since #62, but for the MRs above, nothing happened.

Looking at the worker logs, the event was received and handled (the source-git repo was cloned and the dist-git repo was updated), but handling ended with a `PR already exists: https://gitlab.com/packit-service/rpms/systemd/-/merge_requests/1` message and nothing else was done.

Re-opening the source-git MR produces the same behaviour, that is: there is no update in the dist-git MR.

**Comment by jpopelka** [comment]
The problem was in the database. It's probably related to the fact that you created [src#2](https://gitlab.com/packit-service/src/systemd/-/merge_requests/2) with the same content as [src#1](https://gitlab.com/packit-service/src/systemd/-/merge_requests/1), packit refused to create a different dist-git MR and hardly couldn't add another relation into db, because the [rpms#1](https://gitlab.com/packit-service/rpms/systemd/-/merge_requests/1) was already there in a relation with the src#1.

I fixed it in the database, so it works in this case.  But it's a bug and has to be investigated and fixed.

**Comment by jpopelka** [comment]
Might be related to packit/packit-service#1504

**Comment by jpopelka** [comment]
(At least) two things need to be done here:

* Once packit/packit#1623 is merged we have to change the [condition here](https://github.com/packit/hardly/blob/767fde4128ac251a5cf27f9bfe6e9fa2c95357da/hardly/handlers/distgit.py#L206) because `sync_release` will no longer return `None` (but the PR itself) in case of already existing PR
* We need to handle the case when someone creates another source-git PR with the same title and description. Because now, packit refuses to create another PR. We either have to:
  * add some more info into the dist-git PR description, which distinguishes it from the already existing PR, or
  * add some `check_existing_pr: bool` argument to `sync_release` which would allow us to skip [the checking for already existing PR](https://github.com/packit/packit/blob/7a4ef10509a7dbce2a5fca85d6c8f21350965998/packit/api.py#L781)

**Comment by jpopelka** [comment]
> We need to handle the case when someone creates another source-git PR with the same title and description. Because now, packit refuses to create another PR. We either have to:
>
>     * add some more info into the dist-git PR description, which distinguishes it from the already existing PR

Actually, the dist-git MR description contains a link to the src-git MR so "when someone creates another source-git PR with the same title and description" the dist-git MR description would be different and hence this ("packit refuses to create another PR") could not happen.

I no longer understand how the problem was triggered (as I describe in https://github.com/packit/hardly/issues/70#issuecomment-1137086094) nor how to reproduce it.
* When there's an opened src-git MR and I try to open a new MR from the same branch I get "Another open merge request already exists for this source branch"
* When I first close the src-git MR (which closes the corresponding dist-git MR) and then open a new src-git MR from the same branch, with the same title and description a new dist-git MR is opened just fine because the previous dist-git MR was closed at the time of checking for an already existing dist-git MR.
* When I first close the src-git MR (which closes the corresponding dist-git MR), **then reopen the dist-git MR** and then open a new src-git MR from the same branch, with the same title and description a new dist-git MR is opened just fine because the previous dist-git MR had a different src-git MR URL in the description.

Closing
